src/Ocelot.Provider.Polly/PollyQoSResiliencePipelineProvider.cs

```diff
@@ -39,46 +61,65 @@ public ResiliencePipeline<HttpResponseMessage> GetResiliencePipeline(DownstreamR
         }
 
         var currentRouteName = GetRouteName(route);
-        return _resiliencePipelineRegistry.GetOrAddPipeline<HttpResponseMessage>(
-            key: new OcelotResiliencePipelineKey(currentRouteName), 
+        return _registry.GetOrAddPipeline<HttpResponseMessage>(
+            key: new OcelotResiliencePipelineKey(currentRouteName),
             configure: (builder) => PollyResiliencePipelineWrapperFactory(builder, route));
     }
 
-    private void PollyResiliencePipelineWrapperFactory(ResiliencePipelineBuilder<HttpResponseMessage> builder, DownstreamRoute route)
+    protected virtual void PollyResiliencePipelineWrapperFactory(ResiliencePipelineBuilder<HttpResponseMessage> builder, DownstreamRoute route)
     {
         var options = route.QosOptions;
 
-        // Add TimeoutStrategy if TimeoutValue is not int.MaxValue and greater than 0
-        if (options.TimeoutValue != int.MaxValue && options.TimeoutValue > 0)
+        // Add CircuitBreaker strategy only if ExceptionsAllowedBeforeBreaking is greater than 2
+        if (options.ExceptionsAllowedBeforeBreaking >= 2)
```

Comment:
Why not use the shortcut, `options.ExceptionsAllowedBeforeBreaking < 2` and return?

Reply:
We can't shortcut here because we must care about TimeOut strategy also: see [line 112](https://github.com/ThreeMammals/Ocelot/pull/2086/files#diff-a15546e836c83aa7eaeabbcb1661fb7254214dd72097d83f94611f56fb73a5d8R112)

The workflow is the following: if option exists, apply appropriate Polly strategy.
So, we have 2 major strategies now:
- 1st is CircuitBreaker 
- 2nd is TimeOut

Both of them have appropriate option and `if`-block.

Reply:
ok, maybe, because of the code block size to foresee a method and place the logic in this method.

Reply:
Fixed in commit https://github.com/ThreeMammals/Ocelot/pull/2086/commits/cfcb772d17c607634cee1919dbd717f523634ff2 ✔️